crates/hdx_parser/src/traits/mod.rs

```diff
@@ -154,6 +154,7 @@ pub trait RangedMediaFeature<'a>: Sized {
 	fn new(left: (Comparison, Self::Type), right: Option<(Comparison, Self::Type)>, legacy: bool) -> Self;
 
 	fn parse_ranged_media_feature(name: Atom, parser: &mut Parser<'a>) -> Result<Self> {
+		let checkpoint = parser.checkpoint();
```

Comment:
Is this correct? Set a checkpoint, then call `.next()` and of we don't get the token we want, then we rewind to that checkpoint?

Reply:
yep that's exactly it!